docs: Add new "Tips and Tricks" section

Funky185540 opened the pull request:
Adds a new section containing general tips and tricks for working with
rpm-ostree based operating systems.

@travier Tagging you here due to the sync to the Kinoite docs, in case this gets approval. :)

travier reviewed (approved):
I like those! Thanks

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+= Tips and Tricks
+
+== Working with Toolbox
```
s/toolbox/Toolbx

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+
+== Working with Toolbox
+
+=== Running applications from inside toolbox on the host
```
s/toolbox/Toolbx

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+=== Running applications from inside toolbox on the host
+
+This can be necessary if you want to interact with tools available from the
+host, for example `podman`, `nmcli` or `rpm-ostree` without leaving the toolbox
```
s/toolbox/Toolbx container/

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+
+  $ flatpak-spawn --host sudo -S rpm-ostree status
+
+If you find yourself using commands like these frequently to access e.g. flatpak
```
s/flatpak/flatpak command/

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+  $ flatpak-spawn --host sudo -S rpm-ostree status
+
+If you find yourself using commands like these frequently to access e.g. flatpak
+from inside the toolbox, you can create yourself a short custom wrapper script
```
s/toolbox/Toolbx container/

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+
+If you find yourself using commands like these frequently to access e.g. flatpak
+from inside the toolbox, you can create yourself a short custom wrapper script
+(*inside toolbox*):
```
s/toolbox/Toolbx container/

tpopela commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,87 @@
+
+
+
+== Working with `flatpak`
```
s/`flatpak`/Flatpak applications/

tpopela reviewed (changes requested):
Apart from small nitpicks it looks good to me, once you fix those we can merge it in.

Funky185540 commented:
Thanks for the feedback!

The changes are in now, hope I spotted everything.

tpopela commented:
I think that https://github.com/fedora-silverblue/silverblue-docs/pull/119#discussion_r786687346 is still valid - replace `flatpak` with Flatpak applications

Funky185540 commented:
Sorry, missed those! Should be in now. Besides that, I added a new subsection to working with toolbx in a second commit.

travier commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,119 @@
+flatpak command from inside the Toolbx container, you can create yourself a
+short custom wrapper script (*inside the Toolbx container*):
+
+  $ echo '#!/bin/sh
+    exec /usr/bin/flatpak-spawn --host flatpak "$@"' | \
+    sudo tee /usr/bin/flatpak && sudo chmod +x /usr/bin/flatpak
```
Looks like there is something missing here. A redirect to a file maybe?

Funky185540 replied:
That's what `tee` is there for. I can't just `>` to a file, because I need to be root to write to `/usr/bin`. If I `sudo echo`, the command after the pipe will still be spawned as user process. Alternatively I could also run the whole thing as root and prefix the prompt with `#`, but I find this nicer.

This is also the method you'd usually use to fiddle with e.g. sysfs entries when not working as root all the time.

Funky185540 replied:
But thinking about it, should I add a note that when the user, after executing this command, sees the following reply: `tee: /usr/bin/foobar: Read-only file system` they forgot to enter their toolbx container? It's Silverblue keeping them from shredding their hosts `flatpak` executable. :)

travier replied:
Oh, I had missed the end quote and miss-read the whole thing.
Let's make it one line with `echo -e '...\n...'` and write to `/usr/local/bin` instead?
To make it clear that this has to be run inside a toolbox, you may use `[toolbox]$` for the prompt maybe?

travier commented on modules/ROOT/pages/tips-and-tricks.adoc:
```diff
@@ -0,0 +1,119 @@
+
+[NOTE]
+====
+For this to become available inside your Toolbx containers, they must be
```
For the alias to work? That's weird.

Funky185540 replied:
Well spotted, that's wrong in this context. I previously posted this somewhere else where it was relevant, thanks!

travier reviewed (commented):
Two minor comments.

Funky185540 commented:
> Oh, I had missed the end quote and miss-read the whole thing. Let's make it one line with `echo -e '...\n...'` and write to `/usr/local/bin` instead? To make it clear that this has to be run inside a toolbox, you may use `[toolbox]$` for the prompt maybe?

@travier Thanks for the input. I now turned it into a (lengthy) oneliner and pipe the output of `tee` to /dev/null so it doesn't show up on the CLI, because it's rather confusing. I also added the `[toolbx]` in front of the prompt hoping it reduces confusion and helps someone...

However I kept the path at `/usr/bin/flatpak`, because I think that this is rather nice for the following reason:

- If the user does run it from the host, maybe this reassures them that they cannot (by accident) delete their systems executables under `/usr/bin`, which is a cool thing to have tested once. :)
- If I changed the path to `/usr/local/bin`, then executing the command on the host would actually create the `flatpak` command there, and given that this comes before `/usr/bin` in `$PATH`, the user would always run flatpak as `flatpak-spawn --host flatpak ...` without ever noticing. This feels wrong and I'd like to avoid people running into weird issues with this...
- Also this would allow anyone interested to hardcode `flatpak` e.g. in custom scripts as `/usr/bin/flatpak`, which hopefully should aid in not breaking workflows for people depending on whether they are running on the host or in toolbx...

Not sure if the last one's really relevant, but I could very well imagine so. :)

travier commented:
If we are concerned that people might run this in the wrong environment, then I think we should have the script check for that in order to make it do the right thing in all cases.

I'm not sure writing to `/usr/bin/flatpak` is a good idea as it might create unexpected interactions if users install flatpak inside their toolbx (will they get a clean error from dnf? I have not tried that yet). They might not install it directly but it could get pulled as a dependency.

Thus I think we should keep it at `/usr/local/` and we should probably reverse the order of the first two sections to make it logical:
- Here is how to check if you are running in a toolbox.
- Here is how to make calling a program work whatever the environment you are in.

As this will probably be a much longer line now, maybe we should show it like this instead:

```
Ensure you are running in a toolbx:
$ cat ...

Write the following content to `.....`:
$ cat .....
.....

Don't forget to make it executable:
$ chmod ...
```

Funky185540 commented:
I kept the general structure but reordered the Toolbx subsections as suggested. The CLI for creating the alias is still very long, is that ok?

travier commented:
Ping @tpopela or I'll merge this on Monday 🙂.

Funky185540 commented:
Thank you! :)
